gsa/edits.go

```diff
@@ -14,7 +16,29 @@ package gsa
 //      The two strings without gaps and the list of edit operations
 //      as a string.
 func GetEdits(p, q string) (gapFreeP, gapFreeQ, edits string) {
-	gapFreeP, gapFreeQ, edits = "", "", ""
+	var sb_p, sb_q, sb_edits strings.Builder
+
+	for i := 0; i < len(p); i++ {
+
+		if p[i] == []byte("-")[0] {
```

Comment:
When you write something like `[]byte("-")[0]`, surely your brain is screaming **"IT CAN'T POSSIBLY BE THAT DIFFICULT TO GET A BYTE!!!"** Right? You are creating a string that you turn into a byte slice, where you then get the first character. That can't be right.

Your brain was complaining while you wrote this, wasn't it? That is what a brain should be doing in a situation like that. And unless it is late at night, or after Friday bar, or possibly both, (which the timing of the commit does allow for), that brain should start looking for better solutions.

Now, I freely admit that sometimes things just are that weird, but for basic types, I have yet to see it.

In `Go`, you get a byte by setting a character in single quotes. What you want here is probably:

```go
if p[i] == '-' {
    ...
}
```